backend/src/main/java/com/example/backend/entity/DistanceMeasurable.java

```diff
@@ -0,0 +1,7 @@
+package com.example.backend.entity;
+
+public interface DistanceMeasurable {
+    Location getLocation();
+
+    float distanceTo(DistanceMeasurable other);
```

Comment:
I'd add a docstring for this, but you could argue that this is intuitive enough that it doesn't need it. Up to you.